interfaces.go

```diff
@@ -145,6 +169,28 @@ type Event struct {
 	Exception   []Exception            `json:"exception,omitempty"`
 }
 
+func (e *Event) MarshalJSON() ([]byte, error) {
+	type alias Event
+	// encoding/json doesn't support the "omitempty" option for struct types.
+	// See https://golang.org/issues/11939.
+	// This implementation of MarshalJSON shadows the original Timestamp field
+	// forcing it to be omitted when the Timestamp is the zero value of
+	// time.Time.
+	if e.Timestamp.IsZero() {
+		return json.Marshal(&struct {
+			*alias
+			Timestamp json.RawMessage `json:"timestamp,omitempty"`
```

Comment:
Here we could have used any type for `Timestamp` as long as it works with `omitempty`.

I chose `json.RawMessage` (over `bool`, `[]byte`, `int`, etc) because of the semantic value it carries.
Since this is not exported we can change it later at will.

IIRC, my intention was to use `json.RawMessage` in other fields to add pre-encoded chunks of JSON.